pkg/resources/scim_integration_acceptance_test.go

```diff
@@ -1,79 +1,180 @@
 package resources_test
 
 import (
-	"fmt"
 	"testing"
 
 	acc "github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/acceptance"
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/helpers"
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/internal/snowflakeroles"
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/sdk"
 
+	"github.com/hashicorp/terraform-plugin-testing/config"
 	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
 	"github.com/hashicorp/terraform-plugin-testing/tfversion"
 )
 
-func TestAcc_ScimIntegration(t *testing.T) {
-	scimIntName := acc.TestClient().Ids.Alpha()
-	scimProvisionerRole := "AAD_PROVISIONER"
-	scimNetworkPolicy := acc.TestClient().Ids.Alpha()
+func TestAcc_ScimIntegration_basic(t *testing.T) {
+	networkPolicy, networkPolicyCleanup := acc.TestClient().NetworkPolicy.CreateNetworkPolicy(t)
+	t.Cleanup(networkPolicyCleanup)
 
+	role, roleCleanup := acc.TestClient().Role.CreateRoleWithRequest(t, sdk.NewCreateRoleRequest(snowflakeroles.GenericScimProvisioner).WithOrReplace(true))
```

Comment:
I thought these roles are preexisting, because you have added them as constants, I think that we should not replace the if they are preexisting.

Reply:
They are not guaranteed by snowflake and they may not exist in a deployment. That's why we need to provide them. They are added as constants because role names are listed in enum (check [docs](https://docs.snowflake.com/en/sql-reference/sql/create-security-integration-scim)). So, if a user declares an integration with one of these roles, they need to exist in a deployment.

Reply:
Okay, in such a case, I think that:
1. We should not clean them up as part of the test.
2. We should pre-create them on our environments.
3. We should add a check similar to the one added recently for Quoted identifiers ignore case param (check `EnsureQuotedIdentifiersIgnoreCaseIsSetToFalse`)
4. We should assume in our tests that they exist and be explicit with it in our docs.
5. It would be interesting to check what happens if such a role does not exist (I mean to have acceptance test for it):
  - is it nicely handled on SF side?
  - do we pass these error nicely to the user? if not, should we handle it with a check on the provider side?
  - if we want to test it, let's maybe create only one of these 3 roles, assert that second does not exist, and have a test for the non-existing one?

@sfc-gh-jcieslak I would like your opinion here too.

Reply:
Yeah, It's confusing to have predefined values that you have to create on your own, but we have to deal with it somehow. Having it pre-created to be able to use them without remembering to create them before would be nice addition. I agree with all the points.

Reply:
Done, also removed creating these roles from integration tests. Note that  `AadProvisioner` is not provisioned because it's not used in tests.